src/DataTable/FilterAndSortMenu.js

```diff
@@ -52,9 +55,26 @@ export default class FilterAndSortMenu extends React.Component {
     } else if (ccSelectedFilter === "notEmpty") {
       // manually set filter value (nothing is selected by user)
       filterValToUse = true;
+    } else if (ccSelectedFilter === "isEmpty") {
+      // manually set filter value (nothing is selected by user)
+      filterValToUse = false;
+    } else if (
+      ccSelectedFilter === "inList" ||
+      ccSelectedFilter === "notInList"
+    ) {
+      if (dataType === "number") {
+        if (Array.isArray(filterValue))
+          filterValToUse = filterValue.map(val =>
+            val.includes(",")
+              ? parseFloat(val.replaceAll(",", "")) || ""
+              : parseFloat(val)
+          );
+      }
```

Comment:
If ccSelectedFilter is a list filter, filterValue will always be an array. This could simply be:

```javascript
} else if (
      ccSelectedFilter === "inList" ||
      ccSelectedFilter === "notInList"
  ) {
    if (dataType === "number") {
      filterValToUse = filterValue.map(val => parseFloat(val.replaceAll(",", "")))
    }

```